File: src/modules/company/services/company-delete-avatar.service.ts
import { Injectable, NotFoundException } from '@nestjs/common';
import { PrismaService } from 'src/infra/database/prisma.service';
import { StoredFileDeleteService } from 'src/modules/stored-file/services/stored-file-delete.service';
import { UserRequest } from 'src/shared/types/user-request';
import { CompanyEntity } from '../domain/entities/company.entity';
import { userCompanyValidator } from 'src/shared/validators/user-company.validator';

@Injectable()
export class CompanyDeleteAvatarService {
  constructor(
    private prisma: PrismaService,
    private storedFileDeleteService: StoredFileDeleteService,
  ) {}
  async execute(id: number, userRequest: UserRequest): Promise<CompanyEntity> {
    await userCompanyValidator(id, userRequest);

    const company = await this.prisma.company.findFirst({
      where: {
        id: id,
        deletedAt: null,
      },
      include: {
        avatar: true,
      },
    });

    if (!company) throw new NotFoundException('Arquivo não encontrado!');

    const updatedCompany = await this.prisma.company.update({
      data: {
        avatarId: null,
      },
      where: {
        id: company.id,
      },
      include: {
        avatar: true,
      },
    });

    await this.storedFileDeleteService.execute(company.avatar.uuid);

    return new CompanyEntity(updatedCompany);
  }
}

File: src/modules/company/services/company_delete_avatar_service.py
from src.shared.types.user_request import UserRequest
from src.modules.company.models import Company
from src.shared.exceptions.not_found_exception import NotFoundException
from src.shared.validators.user_company_validator import (
    user_company_validator,
)
from src.modules.stored_file.services.stored_file_delete_service import (
    StoredFileDeleteService,
)


class CompanyDeleteAvatarService:
    @classmethod
    def execute(self, id: int, user_request: UserRequest):

        company = Company.objects.filter(id=id, deleted_at=None).first()

        if company is None:
            raise NotFoundException("Empresa não encontrada!")

        if company.avatar is None:
            raise NotFoundException("Arquivo não encontrado!")

        user_company_validator(id, user_request)

        uuid = company.avatar.uuid

        company.avatar = None
        company.save()

        StoredFileDeleteService.execute(uuid)

        return company
